### PR TITLE
fix/etherscan verification

### PR DESCRIPTION
**Pull request by rkrasiuk**
## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

File: ethers-etherscan/src/contract.rs
Comment (rkrasiuk) [inline ethers-etherscan/src/contract.rs:329]: the parameters for this test will be changed out for the ones exposed by #809

**Comment by mattsse** [comment]
what could be nice to have is some roundtrip test:
1. get contract source code for a specific address
2. get metadata (hopefully compiler version and constructor arguments are included)
3. call verify

**Comment by gakonst** [comment]
@rkrasiuk FYI #809 got rolled in #802 so if you'd want to unblock yourself from it you could use that as the base for this PR. Otherwise, will ping again when 802 is merged and we can then move forward with it?

**Comment by rkrasiuk** [comment]
@gakonst what do you think about the following - I add a roundtrip test and we merge as it is. then, in foundry, we require to provide compiler version and num of optimizations for now. then, after #802 is merged, we just improve on top of this PR

**Comment by gakonst** [comment]
agreed - sgtm

**Comment by rkrasiuk** [comment]
@gakonst perfect, will take care of this tmrw morning

**Comment by rkrasiuk** [comment]
@mattsse so, the metadata is all there in the response. the problem with the roundtrip test is that etherscan would error out on trying to verify the contract that was already verified. i guess we can redeploy the contract each time on goerli for verification, but we'd have to set up some dummy wallet with goerli eth

**Comment by mattsse** [comment]
oh I see, I don't think that's worth it.
looking good!

**Review by gakonst — APPROVED** [review]
Looks good! Let's maybe look into how we can expand flattening to support multiple contracts In the same file?

**Pull request merged**
